File: mobile/lib/features/feedback/domain/entities/feedback.dart
enum FeedbackTopic {
  general(0, 'Trải nghiệm chung'),
  bookingOrder(1, 'Đặt tủ / đơn hàng'),
  payment(2, 'Thanh toán'),
  delivery(3, 'Giao nhận'),
  interface(4, 'Giao diện'),
  other(5, 'Khác');

  const FeedbackTopic(this.value, this.label);

  final int value;
  final String label;

  static FeedbackTopic fromValue(int value) {
    return FeedbackTopic.values.firstWhere(
      (topic) => topic.value == value,
      orElse: () => FeedbackTopic.general,
    );
  }
}

class UserFeedback {
  const UserFeedback({
    required this.id,
    required this.rating,
    required this.topic,
    required this.content,
    required this.pageUrl,
    required this.isVisible,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final int rating;
  final FeedbackTopic topic;
  final String content;
  final String pageUrl;
  final bool isVisible;
  final DateTime createdAt;
  final DateTime updatedAt;
}

File: backend/src/Locker.Backend.Domain/Entities/Feedback.cs
using Locker.Backend.Domain.Enums;

namespace Locker.Backend.Domain.Entities;

public class Feedback : BaseEntity
{
    public Guid UserId { get; set; }
    public string Username { get; set; } = string.Empty;
    public int Rating { get; set; }
    public FeedbackTopic Topic { get; set; }
    public string Content { get; set; } = string.Empty;
    public string PageUrl { get; set; } = string.Empty;
    public bool IsVisible { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
